R/environment.r

```diff
@@ -0,0 +1,18 @@
+# Everthing related to the environment which takes functions which shadow base R functions.
+# This is needed to build in our own needs, like properly shutting down the kernel
+# when `quit()` is called.
+
+# The real env is created and attach'ed in the main() via the call to init_shadowenv()!
+init_shadowenv <- function(){
+    .irk.shadowenv <- attach(NULL, name = "jupyter:irkernel")
+
+    add_to_user_searchpath <- function(name, FN, attrs = list()) {
+        assign(name, FN, .irk.shadowenv)
+    }
+
+    # add the accessors to the shadow env to that env, so they are actually accessable 
+    # from everywhere...
+    add_to_user_searchpath(".irk.get_shadowenv", function() {.irk.shadowenv})
```

Comment:
That's another "habit" (apart from the smartquotes... :-)) of mine which I don't want to break: all codeblocks belong into brackets so that in the next "enter", the functions still works and does not need to be edited to get such brackets.

-> If there is error to have these brackets here, I would like to keep them...